dali/operators/reader/loader/loader.h

```diff
@@ -217,8 +217,8 @@ class Loader {
   void PrepareMetadata() {
     std::lock_guard<std::mutex> l(prepare_metadata_mutex_);
```

Comment:
Perhaps instead of checking this flag outside, it should be checked here and the `if (!loading_flag)` removed from all usages.
```C++
if (loading_flag_)
  return;
```

Reply:
Done